File: python/selection-sort.py
def selection_sort(arr):
    for i in range(len(arr)-1):
        min_index =i
        for j in range(i+1,len(arr)):
            if arr[j] < arr[min_index]:
                min_index=j

        if i!=min_index:
            temp=arr[i]
            arr[i]=arr[min_index]
            arr[min_index]=temp



arr=[2,34,1,25,7,8]
print(arr , end="\n")
selection_sort(arr)
print(arr)

File: cpp/sorting/selectionSort.cpp
#include <iostream>
using namespace std;

void SelectionSort(int arr[], int n) {
  for (int i = 0; i < n - 1; i++) {
    for (int j = i + 1; j < n; j++) {
      if (arr[j] < arr[i]) {
        int temp = arr[j];
        arr[j] = arr[i];
        arr[i] = temp;
      }
    }
  }
  for (int i = 0; i < n; i++) {
    cout << arr[i] << " ";
  }
}

int main27() {
  int n = 6;
  int arr[6] = {12, 32, 2, 21, 66, 1};
  SelectionSort(arr, n);
}